surprise-pairwise-from-embedding.R

```diff
@@ -0,0 +1,151 @@
+        all_sim[[i]] <- 0}
+    } #end of source loop
+    max_sim[[j]] <- max(unlist(all_sim)) #the maximum of the calculated similarities
+  }#end of target loop
+  target$max_sim <- unlist(max_sim) 
+  target1 <- target %>% filter(max_sim > 0)
+  
+  # add weight by act (the diff between act_mean and all_mean)
+  # "diff" from surprise_linear-correlation-time.R
+  
+  max_sim <- list()
+  for(l in 1:nrow(target1)){  
+    if (as.numeric(target1[l,2]) %in%  as.numeric(act_length[1,3]):as.numeric(act_length[2,3])){
+      max_sim[[l]] <- target1[l,8] + diff[[1]] } # 8th column if ACT is added before
+    
+    if (as.numeric(target1[l,2]) %in%  as.numeric(act_length[2,3]):as.numeric(act_length[3,3])){
+      max_sim[[l]] <- target1[l,8] + diff[[2]] }
+    
+    if (as.numeric(target1[l,2]) %in%  as.numeric(act_length[3,3]):as.numeric(act_length[4,3])){
+      max_sim[[l]] <- target1[l,8] + diff[[3]] }
+    
+    if (as.numeric(target1[l,2]) %in%  as.numeric(act_length[4,3]):as.numeric(act_length[5,3])){
+      max_sim[[l]] <- target1[l,8] + diff[[4]] }
+    
+    if (as.numeric(target1[l,2]) %in%  as.numeric(act_length[5,3]):as.numeric(act_length[6,3])){
+      max_sim[[l]] <- target1[l,8] + diff[[5]] }
+  }
+  
+  pair_sim [[k]] <- tibble(
+    source = pairs[k,1],
+    target = pairs[k,2],
+    flow = mean(unlist(max_sim))
+  ) #end of pairs loop 
+}
+
+# summarise everything in one tibble
+pairwise_flow <- bind_rows(pair_sim)
+
+# add network normalization to the flow between two chars:
+#first calculate all flow to one char as target, than divide the actual measure when this char is a source
+
+# all flow to one char
+flow_to_char <- pairwise_flow %>% 
+  group_by(target) %>% 
+  group_split()
+
+flow_to_char1 <- list()
+for (i in 1:length(flow_to_char)){
+  flow_to_char1[[i]] <- mean(flow_to_char[[i]]$flow)}
+
+flow_to_all <- tibble(flow_to = unlist(flow_to_char1), char = names(chars))
+
+#add to dataframe and divide the flow when char is source
+pairwise_weigthed <- merge(pairwise_flow, flow_to_all, by.x = "source", by.y = "char")
+pairwise_weigthed <- pairwise_weigthed %>% 
+  mutate(weigthed_flow = flow/flow_to)
+
+## calculate difference beetween S-T T-S pairs
+pair_diff <- list()
+for (i in 1:nrow(pairwise_weigthed)){
+  for( j in 1:nrow(pairwise_weigthed)){
+    if (pairwise_weigthed[i,1] == pairwise_weigthed[j,2] & pairwise_weigthed[j,1] == pairwise_weigthed[i,2]){
+      pair_diff [[i]] <-
+        tibble(
+          source = pairwise_weigthed[i,1],
+          target = pairwise_weigthed[i,2],
+          flow_w = pairwise_weigthed[i,5]-pairwise_weigthed[j,5],
+          flow = pairwise_weigthed[i,3]-pairwise_weigthed[j,3]
+        )}}}
+
+pair_diff <- bind_rows(pair_diff)
+pair_diff <- pair_diff %>% 
+  filter(flow > 0)
+colnames(pair_diff) <- c("source", "target", "flow_w", "flow")
+
+#save results
+capture.output(as.data.frame(pair_diff), file = "pairwise-cosine-longsentence-wiegtbyact.csv")

```

Comment:
I have never seen capture.output used on as.data.frame to save a csv, and I don't really see how it could work. Shouldn't this just be `readr::write_csv(pair_diff, "pairwise-cosine-longsentence-wiegtbyact.csv")`